### PR TITLE
[ci] Add API Scan job

### PR DESCRIPTION
**Pull request by pjcollins**
Context: https://devdiv.visualstudio.com/DevDiv/_wiki/wikis/DevDiv.wiki/25351/APIScan-step-by-step-guide-to-setting-up-a-Pipeline

The ApiScan task has been added to pipeline runs against `main`.  This
task should help us identify related issues earlier, rather than having
to wait for a full scan of VS.

### CODE REVIEW

**Comment by pjcollins** [comment]
Api Scan test run - https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=8898601&view=results

File: azure-pipelines.yml [hunk 1]
```diff
@@ -308,7 +308,15 @@ stages:
...
           archiveFilePatterns: $(Build.StagingDirectory)\**\*.nupkg
-          destinationFolder: $(Build.StagingDirectory)\apiscan
+          destinationFolder: $(Build.SourcesDirectory))\nuget
+
+      - task: CopyFiles@2
+        displayName: Collect Files for APIScan
+        inputs:
+          Contents: |
+            $(Build.SourcesDirectory))\nuget
+            !$(Build.SourcesDirectory)\**\runtimes\win-arm64\native\libzipsharpnative*.dll
```
[the remark this answers is not in the document]
Comment (pjcollins) [reply azure-pipelines.yml:318]: The scan is triggered by .dll and .exe files by my understanding so I think skipping the copy of invalid assemblies should be enough.

**Comment by pjcollins** [comment]
The recommended method for suppressing known API Scan issues is to wait for the requested server exception list (SEL) changes to be approved and updated. This should be good to go with that in mind, but we may want to wait to merge until after the SEL is updated so that the new job will pass.

**Comment by pjcollins** [comment]
Latest API scan run after 1ES template migration and rebasing: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=9129458&view=results

**Comment by pjcollins** [comment]
Latest test run trying to match VS server exception list values: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=9142482&view=results

**Review by dellis1972 — APPROVED** [review]

**Review by dellis1972 — APPROVED** [review]

**Review by grendello — APPROVED** [review]

**Pull request merged**